app/src/views/search-engine/components/search-bar/index.js

```diff
@@ -25,11 +25,23 @@ const SearchBar = ({ setSearch, refetch }) => {
               root: classes.inputRoot,
               input: classes.inputInput
             }}
-            onChange={({ target: { value } }) => setSearch(value)}
+            onChange={({ target: { value } }) => {
+              if (value?.length !== 0) {
+                setLoading(true)
+                setSearch(value)
```

Comment:
What happens if you setSearch when value?.length == 0? I think it would be nicer code to have this as
`setLoading(value?.length !== 0)`
`setSearch(value)`
or possibly validate the condition in setSearch

Reply:
Yes, it actually does look nicer, but if you notice, I am trying to avoid calling `setSearch()` if the value is an empty string. Whenever `setSearch()` gets invoked, a network request get queued, and we want to avoid that if `value` is empty.